onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -139,42 +139,50 @@ common::Status OpenVINOExecutionProvider::Compile(
-                                                      graph_body_viewer,
-                                                      *GetLogger(),
-                                                      ep_ctx_handle_);
-
-    compute_info.create_state_func =
-        [backend_manager](ComputeContext* context, FunctionState* state) {
-          OpenVINOEPFunctionState* p = new OpenVINOEPFunctionState();
-          p->allocate_func = context->allocate_func;
-          p->destroy_func = context->release_func;
-          p->allocator_handle = context->allocator_handle;
-          p->backend_manager = backend_manager;
-          *state = static_cast<FunctionState>(p);
-          return 0;
-        };
-    compute_info.compute_func = [](FunctionState state, const OrtApi* /* api */, OrtKernelContext* context) {
-      auto function_state = static_cast<OpenVINOEPFunctionState*>(state);
-      try {
-        function_state->backend_manager->Compute(context);
-      } catch (const std::exception& ex) {
-        return common::Status(common::ONNXRUNTIME, common::FAIL, ex.what());
+    try {
+      std::shared_ptr<openvino_ep::BackendManager> backend_manager =
+          std::make_shared<openvino_ep::BackendManager>(*global_context_,
+                                                        fused_node,
+                                                        graph_body_viewer,
+                                                        *GetLogger(),
+                                                        ep_ctx_handle_);
+      compute_info.create_state_func =
+          [backend_manager](ComputeContext* context, FunctionState* state) {
+            OpenVINOEPFunctionState* p = new OpenVINOEPFunctionState();
+            p->allocate_func = context->allocate_func;
+            p->destroy_func = context->release_func;
+            p->allocator_handle = context->allocator_handle;
+            p->backend_manager = backend_manager;
+            *state = static_cast<FunctionState>(p);
+            return 0;
+          };
+      compute_info.compute_func = [](FunctionState state, const OrtApi* /* api */, OrtKernelContext* context) {
+        auto function_state = static_cast<OpenVINOEPFunctionState*>(state);
+        try {
+          function_state->backend_manager->Compute(context);
+        } catch (const std::exception& ex) {
+          return common::Status(common::ONNXRUNTIME, common::FAIL, ex.what());
+        }
+        return Status::OK();
+      };
+
+      compute_info.release_state_func =
+          [](FunctionState state) {
+            if (state) {
+              OpenVINOEPFunctionState* function_state = static_cast<OpenVINOEPFunctionState*>(state);
+              delete function_state;
+            }
+          };
+      node_compute_funcs.push_back(compute_info);
+    } catch (const OnnxRuntimeException& ex) {
+      std::string exception_str = ex.what();
+      if (exception_str.find("ZE_RESULT_ERROR_UNKNOWN") != std::string::npos ||
```

Comment:
Sriram, Mayuresh pointed out that all applications need to be caught
Not just these two

Reply:
And the string need to be appended with error code